cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-video.php

```diff
@@ -338,16 +338,55 @@ public function print_video_scripts() {
 				}
 
 				$config = wp_parse_args( $video['args'], $default );
-				// Sizing.
-				if ( empty( $config['size'] ) ) {
+
+				if ( empty( $config['size'] ) && ! isset( $this->config['video_freeform'] ) ) {
 					$config['fluid'] = true;
 				}
-				$code[] = 'cld.videoPlayer(\'cloudinary-video-' . $instance . '\', ' . wp_json_encode( $config ) . ');';
+
+				$cld_videos[ $instance ] = $config;
+
 			}
-			// If code was populated, output.
-			if ( ! empty( $code ) ) {
-				wp_add_inline_script( 'cld-player', implode( $code ) );
+
+			if ( empty( $cld_videos ) ) {
+				return;
 			}
+
+			ob_start();
+			?>
+			var cldVideos = <?php echo wp_json_encode( $cld_videos ); ?>;
+
+for ( var videoInstance in cldVideos ) {
```

Comment:
Ah, I meant indenting this entire script to the same level. What do you think?

Like:

```php
	public function print_video_scripts() {

		if ( $this->player_enabled() && ! empty( $this->attachments ) ) {

			$cld_videos = array();
			foreach ( $this->attachments as $instance => $video ) {
				// @todo - ping the URL to ensure it has transformation available, else update an eager.
				$cloudinary_id = $this->media->get_public_id( $video['id'] );
				$default       = array(
					'publicId'    => $cloudinary_id,
					'sourceTypes' => array( $video['format'] ), // @todo Make this based on eager items as mentioned above.
					'controls'    => 'on' === $this->config['video_controls'] ? true : false,
					'autoplay'    => 'off' !== $this->config['video_autoplay_mode'] ? true : false,
					'loop'        => 'on' === $this->config['video_loop'] ? true : false,
				);

				$valid_autoplay_modes = [ 'never', 'always', 'on-scroll' ];
				if ( $default['autoplay'] && in_array( $this->config['video_autoplay_mode'], $valid_autoplay_modes, true ) ) {
					$default['autoplayMode'] = $this->config['video_autoplay_mode'];
				}

				$config = wp_parse_args( $video['args'], $default );

				if ( empty( $config['size'] ) && ! isset( $this->config['video_freeform'] ) ) {
					$config['fluid'] = true;
				}

				$cld_videos[ $instance ] = $config;

			}

			if ( empty( $cld_videos ) ) {
				return;
			}

			ob_start();
			?>
			var cldVideos = <?php echo wp_json_encode( $cld_videos ); ?>;
			
			for ( var videoInstance in cldVideos ) {
				var config = cldVideos[ videoInstance ];
				var id = 'cloudinary-video-' + videoInstance;
				cld.videoPlayer( id, config );
			}
			
			<?php if ( $this->config['video_freeform'] ): ?>
			window.addEventListener( 'load', function() {
				for ( const instance in cldVideos ) {
					var config = cldVideos[ instance ];
					var id = 'cloudinary-video-' + instance;
					var videoContainer = document.getElementById( id );
					var videoElement = videoContainer.getElementsByTagName( 'video' );
			
					if ( videoElement.length === 1 ) {
						videoElement = videoElement[0];
						videoElement.src = videoElement.src.replace( 
							'upload/', 
							'upload/<?php echo esc_js( $this->config['video_freeform'] ) ?>' 
						);
					}
				}
			} );
			<?php 
			endif; 
			
			$script = ob_get_clean();

			wp_add_inline_script(
				'cld-player',
				$script
			);
		}
	}
```